addons/hw_posbox_homepage/controllers/main.py

```diff
@@ -256,13 +256,16 @@ def clear_handlers_list(self):
     @http.route('/server_connect', type='http', auth='none', cors='*', csrf=False)
     def connect_to_server(self, token, iotname):
         if token:
-            credential = token.split('|')
-            url = credential[0]
-            token = credential[1]
-            db_uuid = credential[2]
-            enterprise_code = credential[3]
+            # credential = token.split('|')
+            # url = credential[0]
+            # token = credential[1]
+            # db_uuid = credential[2]
+            # enterprise_code = credential[3]
             try:
-                helpers.save_conf_server(url, token, db_uuid, enterprise_code)
+                configuration = helpers.parse_url(token)
+                helpers.save_conf_server(**configuration)
+            except ValueError:
+                return 'Invalid URL provided.'
```

Comment:
maybe add a _logger.exception on here with the token provided ? (except if it is expected to happen)

Reply:
good idea